writers/batchwriter/batchwriter.go

```diff
@@ -145,7 +148,8 @@ func (w *BatchWriter) worker(ctx context.Context, tableName string, ch <-chan *m
 				w.flushTable(ctx, tableName, resources)
 				resources, sizeBytes = resources[:0], 0
 			}
-			tick = timer(w.batchTimeout)
+			tickClose()
```

Comment:
you shouldn't need to manually close on every iteration

Reply:
incoming commit:

```go
type ticker interface {
	Chan() <-chan time.Time
	Reset(time.Duration)
	Stop() bool
}
```